MaterialSkin/HTML/material/html/js/browse-functions.js

```diff
@@ -782,23 +782,39 @@ function browseGetRoles(view, curitem, ignoreRoles) {
                 if (undefined!=ignoreRoles && ignoreRoles.has(rid)) {
                     continue;
                 }
+                let roleIds = ALBUM_ARTIST_ROLE==rid || ARTIST_ROLE==rid ? '1,5' : rid;
+                let params = [ARTIST_ALBUM_TAGS, SORT_KEY+ARTIST_ALBUM_SORT_PLACEHOLDER, curitem.id, 'role_id:'+roleIds];
+                browseAddLibId(view, params);
                 if (rid>=20) {
                     let udr = lmsOptions.userDefinedRoles[rid];
                     if (undefined!=udr) {
-                        let params = [ARTIST_ALBUM_TAGS, SORT_KEY+ARTIST_ALBUM_SORT_PLACEHOLDER, curitem.id, 'role_id:'+rid];
-                        browseAddLibId(view, params);
                         actions.push({title:udr.name, svg:'role-'+udr.role, do:{ command: ['albums'], params: params}, weight:81, stdItem:STD_ITEM_ARTIST, udr:rid});
                     }
                 } else {
-                    if (COMPOSER_ARTIST_ROLE==rid) {
-                        var params = [SORT_KEY+TRACK_SORT_PLACEHOLDER, PLAYLIST_TRACK_TAGS, curitem.id, 'role_id:'+rid, 'material_skin_artist:'+curitem.title, 'material_skin_compositions:1'];
-                        browseAddLibId(view, params);
-                        actions.push({title:i18n('Compositions'), svg:'composer', do:{ command: ['tracks'], params: params}, weight:81, stdItem:STD_ITEM_COMPOSITION_TRACKS, udr:COMPOSER_ARTIST_ROLE});
-                    } else if ((TRACK_ARTIST_ROLE==rid && undefined!=ignoreRoles) || BAND_ARTIST_ROLE==rid) {
-                        let params = [ARTIST_ALBUM_TAGS, SORT_KEY+ARTIST_ALBUM_SORT_PLACEHOLDER, curitem.id, 'role_id:'+rid];
-                        browseAddLibId(view, params);
-                        actions.push({title:TRACK_ARTIST_ROLE==rid ? i18n('Appearances') : i18n('Band/orchestra'), svg:TRACK_ARTIST_ROLE==rid ? 'artist' : 'role-band', do:{ command: ['albums'], params: params}, weight:81, stdItem:STD_ITEM_ARTIST, udr:rid});
+                    let title = '';
+                    let svg = '';
+                    if (TRACK_ARTIST_ROLE==rid) {
```

Comment:
"ignoreRoleIds" is set to the "role_id" value(s) from the command used to create the current listing of artists' releases. The idea being that if showing "Guitarist" role, there is no need for that in the menu. But, if the user is using a single "Artists" browse mode, then no "role_id" is initially specified. So, with your changes "Appearances" would always be shown. This does not sound correct to me, hence my (TRACK_ARTIST_ID==rid && undefined!=ignoreRoles) was trying to avoid that. (But thinking about this now it should also check && !ignoreRoles.isEmpty)